stl/inc/format

```diff
@@ -2650,6 +2658,48 @@ struct _Arg_formatter {
     }
 };
 
+
+// Special compile time version of _Parse_format_specs. This version is parameterized on
+// the type of the argument associated with the format specifier, since we don't really
+// care about avoiding code bloat for code that never runs at runtime, and we can't form
+// the erased basic_format_args structure at compile time.
+template <typename _Ty, typename _ParseContext>
+constexpr typename _ParseContext::iterator _Compile_time_parse_format_specs(_ParseContext& _Pc) {
+    using _CharT                = typename _ParseContext::char_type;
+    using _Context              = basic_format_context<_Fmt_it, _CharT>;
+    using _ArgTraits            = _Format_arg_traits<_Context>;
+    using _FormattedTypeMapping = typename _ArgTraits::template _Storage_type<_Ty>;
+    // If the type is going to use a custom formatter we should just use that,
+    // instead of trying to instantiate a custom formatter for it's erased handle
+    // type
+    using _FormattedType = conditional_t<is_same_v<_FormattedTypeMapping, typename basic_format_arg<_Context>::handle>,
+        _Ty, _FormattedTypeMapping>;
+    auto _Formatter      = formatter<_FormattedType, _CharT>();
+    return _Formatter.parse(_Pc);
+}
+
+// set of format parsing actions that
+// only checks for validity
+template <class _CharT, typename... _Args>
+struct _Format_checker {
+    using _Pc        = basic_format_parse_context<_CharT>;
+    using _ParseFunc = typename _Pc::iterator (*)(_Pc&);
+
+    static constexpr size_t _Num_args = sizeof...(_Args);
+    _Pc _Parse_context;
+    _ParseFunc _Parse_funcs[_Num_args > 0 ? _Num_args : 1];
+
+
+    explicit constexpr _Format_checker(basic_string_view<_CharT> _Fmt) noexcept
+        : _Parse_context(_Fmt, _Num_args), _Parse_funcs{&_Compile_time_parse_format_specs<_Args, _Pc>...} {}

```

Comment:
This is once using `()` and once `{}`

Reply:
that's because the array uses aggregate initialization, you can't initialize an array like this in an initializer list using ()s, ({}) might work, but I prefer {}